### DISCUSSION TITLE
Duvida ToSqlQuery

### DISCUSSION THREAD

**Discussion by allex-ribeiroh** [balta-io/forum#45] [Tira Dúvidas] [answered]
Balta, estou tentando fazer um consulta parecida com a que você passou na aula de EF, porem eu quero que conte as postagens de uma determinada categoria criei conforme abaixo:

/*Classe*/
public class PostSQL
    {
        public string Nome { get; set; }
        public int Qtde { get; set; }
        public Category Category { get; set; }
    }

/*Config Builder*/
modelBuilder.Entity<PostSQL>(x =>
            {
                x.ToSqlQuery(@" select 
	                                Title Nome,
	                                count(Id) Qtde
                                from 
	                                Post
                                group by
	                                Title");
            });

/*Chamada*/
var pSql = context.PostSQLs
                  .AsNoTracking()
                  .Where(x => x.Category.Id == 1)
                  .ToList();

foreach (var item in pSql)
{
    Console.WriteLine($"O post selecionado é {item.Nome}");
}

O erro foi esse abaixo dizendo que não tem uma chave primaria definida:
System.InvalidOperationException: 'The entity type 'PostSQL' requires a primary key to be defined. If you intended to use a keyless entity type, call 'HasNoKey' in 'OnModelCreating'. For more information on keyless entity types, see https://go.microsoft.com/fwlink/?linkid=2141943.'

Pode orientar por favor?

Grato
Alexandre

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde, @allex-ribeiroh sua classe `public class PostSQL` não tem uma chave definida.
Se isto realmente for verdade, você tem que usar o atributo `HasNoKey` ou definir que não há uma chave no mapeamento.
